Add stratis_dbusquery_version rust program

bgurney-rh opened the pull request:
Related: stratis-storage/project#203
Related: #2155

mulkieran commented on src/bin/stratis_dbusquery_version.rs:
```diff
@@ -0,0 +1,30 @@
+use lazy_static::lazy_static;
+use std::time::Duration;
+
+pub const DBUS_PROPS_IFACE: &str = "org.freedesktop.DBus.Properties";
```
Please drop "pub" modifiers on all constants, since we won't and can't export them from here.

bgurney-rh replied:
Resolved.

bgurney-rh commented:
Rebased.

mulkieran commented on src/bin/stratis_dbusquery_version.rs:
```diff
@@ -4,9 +4,8 @@ use dbus::{
-use std::time::Duration;
 use semver::Version;
-
+use std::time::Duration;
```
Please put ```use std::``` above all the other imports.

bgurney-rh commented on developer_tools/stratis_migrate_symlinks.sh:
```diff
@@ -0,0 +1,61 @@
+			then
+				echo "Targets do not match; sending change event to re-synchronize..."
+				udevadm test --action=change /sys/class/block/$tgtbase 1>/dev/null 2>&1
+				rm -fv $devname
```
Yes; I see what you mean.  This is going to be in an upcoming commit, among a number of other changes in the script, which I made, and just tested with a successful link migration run.

bgurney-rh commented:
Here's the output from a successful test run, in case we want to keep or adjust any of the `println!()` statements in the stratis_dbusquery_version program.

```
# ./stratis_migrate_symlinks.sh 
Variant(
    "2.2.0",
)
verparse: Version {
    major: 2,
    minor: 2,
    patch: 0,
    pre: [],
    build: [],
}
STRATIS_VER_UDEV_SYMLINK: Version {
    major: 2,
    minor: 2,
    patch: 0,
    pre: [],
    build: [],
}
Link name: /stratis/spool1/sfs1
Link target: /dev/dm-5
Target base name: dm-5
Future name: /dev/stratis/spool1/sfs1
No future link found; sending change event...
removed '/stratis/spool1/sfs1'
removed directory '/stratis/spool1'
removed directory '/stratis/'
```

bgurney-rh commented:
Also, here's the output from a "failed run", when there was no stratisd process running (right after the "dnf install" command of the newer package):
```
# ./stratis_migrate_symlinks.sh 
Could not obtain version from stratisd: vertest_err: "The name is not activatable"
Attempt to query stratisd version over dbus failed.
```
After running "systemctl start stratisd", it's able to contact the daemon, and runs successfully.

bgurney-rh commented on src/bin/stratis_dbusquery_version.rs:
```diff
@@ -0,0 +1,60 @@
+fn run() -> Result<Version, String> {
+    let vertest = get_version().map_err(|vertest_err| format!("vertest_err: {}", vertest_err))?;
+    println!("{:#?}", vertest);
+    let verparse = Version::parse(
```
Okay; I'm going to add a commit called "adjust version parse error messages"; let me know if those messages are heading in the right direction.

mulkieran replied:
Those messages look good to me!

bgurney-rh commented on src/bin/stratis_dbusquery_version.rs:
```diff
@@ -0,0 +1,60 @@
+        .map(|r: (GetVerRet,)| r.0)?)
+}
+
+fn run() -> Result<Version, String> {
```
Done.

bgurney-rh commented on src/bin/stratis_dbusquery_version.rs:
```diff
@@ -0,0 +1,60 @@
+}
+
+fn run() -> Result<Version, String> {
+    let vertest = get_version().map_err(|vertest_err| format!("vertest_err: {}", vertest_err))?;
```
Resolved.

bgurney-rh commented:
Updated output for a successful test run:
```
# ./stratis_migrate_symlinks.sh 
Link name: /stratis/spool1/sfs1
Link target: /dev/dm-5
Target base name: dm-5
Future name: /dev/stratis/spool1/sfs1
No future link found; sending change event...
removed '/stratis/spool1/sfs1'
removed directory '/stratis/spool1'
removed directory '/stratis/'
```

And a run with the stratisd daemon not running at time of execution:
```
# ./stratis_migrate_symlinks.sh
Could not obtain version from stratisd: vertest_err: "The name is not activatable"
Attempt to query stratisd version over dbus failed.
```
